lib/ansible/plugins/connection/guestfs.py

```diff
@@ -0,0 +1,215 @@
+
+    def _find_python(self):
+        ''' find what python to use '''
+        self._python = None
+        for p in ('python', 'python3'):
+            try:
+                python_version = self.guestfs.sh('%s -V' % p)
+                self._python = p
+                break
+            except RuntimeError:
+                pass
+
+        if self._python is None:
+            raise AnsibleError("No python found on the image, aborting")
+
+    def _mount_tmp(self):
+        ''' mount a tmpfs over /tmp or /run to store various artefacts '''
+        self._tmp = None
+        for tmp in ('/run', '/tmp'):
+            if self.guestfs.is_dir(tmp):
+                self.guestfs.sh("mount -t tmpfs -o size=4M tmpfs %s" % tmp)
+                self._tmp = tmp
+                break
+
+        if self._tmp is None:
+            raise AnsibleError("Cannot mount tmpfs, aborting")
+
+    def _enable_selinux(self):
+        ''' check if we can load a policy and load it '''
+        # man pages indicate this may not work on older system, not sure
+        #how old they have to be
+        if self.guestfs.is_file('/usr/sbin/load_policy'):
+            self.guestfs.sh('/usr/sbin/load_policy')
+
+    def _start_guestfs(self):
+        ''' start the guestfs appliance '''
+        self.guestfs.set_network(True)
+        self.guestfs.set_selinux(True)
+        self.guestfs.launch()
+        # code taken from the manpages
+        roots = self.guestfs.inspect_os()
+        if len(roots) != 1:
+            # TODO handle the case a bit better ?
+            raise AnsibleError("%s has more than one OS,"
+                               "aborting" % self.disk)
+
+        mps = self.guestfs.inspect_get_mountpoints(roots[0])
+
+        def compare(a, b):
+            return len(a) - len(b)
+
+        for device in sorted(mps.keys(), compare):
+            try:
+                self.guestfs.mount(mps[device], device)
+            except RuntimeError as msg:
+                print("%s (ignored)" % msg)
+        self._enable_selinux()
+
+    def _connect(self):
+        ''' connect to the disk image '''
+        super(Connection, self)._connect()
+        if not self._connected:
+            self._start_guestfs()
+            self._mount_tmp()
+            self._find_python()
+            self.guestfs.write(self._script_name(), EXECUTION_SCRIPT.format(self._tmp))
+
+            self._connected = True
+
+    # no need to randomize the filename since that's on a controlled
+    # tmpfs, with a offline system
+    def _script_name(self):
+        ''' return the filename for the execution script '''
+        return '%s/script' % self._tmp
+
+    def _cmd_name(self):
+        ''' return the filename for the command to execute '''
+        return '%s/cmd' % self._tmp
+
+    def _cmd_result_name(self):
+        ''' return the filename for the command output '''
+        return '%s/cmd.out' % self._tmp
+
+    def exec_command(self, cmd, in_data=None, sudoable=False):
```

Comment:
I am not sure to understand. I suspect I need to change sudoable to True rather than False, becuase that seems to be the default everywhere else, but what class variable do I need to update ?